pkg/controller/service.go

```diff
@@ -273,6 +288,14 @@ func (s *ShadowServiceManager) getMappedPort(stateTable PortMapper, svcName, svc
 	return port, nil
 }
 
+func isPortSuitable(trafficType string, sp corev1.ServicePort) bool {
+	if trafficType == annotations.ServiceTypeUDP {
+		return sp.Protocol == corev1.ProtocolUDP
+	}
+
+	return sp.Protocol == corev1.ProtocolTCP
```

Comment:
I think it's better to return `false` here and to add a case for `ServiceTypeTCP` && `ServiceTypeHTTP`.

Reply:
I did it this way to handle both `SeviceTypeHTTP` and `ServiceTypeTCP`. I can write it stricter like this:

```
if trafficType == annotations.ServiceTypeUDP {
  return return sp.Protocol == corev1.ProtocolUDP
}
if trafficType == annotations.ServiceTypeTCP || trafficType == annotations.ServiceTypeHTTP {
  return sp.Protocol == corev1.ProtocolTCP
}

return false
```